lib/watirspec/server.rb

```diff
@@ -0,0 +1,137 @@
+require "socket"
+require 'watirspec/server/app'
+
+module WatirSpec
+  class Server
+    class << self
+      def run_async
+        if WatirSpec.platform == :java
+          Thread.new { run! }
+          sleep 0.1 until WatirSpec::Server.running?
+        elsif WatirSpec.platform == :windows
+          # FIXME: this makes use lose implementation-specific routes!
+          run_in_child_process
+          sleep 0.5 until listening?
+        else
+          pid = fork { run! }
+          sleep 0.5 until listening?
+        end
+
+        if pid
+          # is this really necessary?
+          at_exit {
+            begin
+              Process.kill 0, pid
+              alive = true
+            rescue Errno::ESRCH
+              alive = false
+            end
+
+            Process.kill(9, pid) if alive
+          }
+        end
+      end
+
+      def run!
+        server = TCPServer.new(bind, port)
+        @running = true
+        loop do
+          client = server.accept
+
+          if header = client.gets
+            method, path = header.split
+
+            headers = {}
+            while line = client.gets.split(' ', 2)
+              break if line[0] == ''
+              headers[line[0].chop] = line[1].strip
+            end
+
+            data = nil
+            if method == 'POST'
+              data = client.read(Integer(headers['Content-Length']))
+            end
+
+            status, headers, body = app.response(path, data)
+
+            headers['Server'] = 'ruby'
+            headers = headers.to_a.map { |array| array.join(': ') }
+
+            response = ["HTTP/1.1 #{status}"] + headers
+            response = response.join("\r\n")
+
+            client.puts(response)
+            client.puts("\r\n\r\n")
+            client.puts(body)
+          end
+
+          client.close
+        end
+      end
+
+      def app
+        @app ||= App.new
+      end
+
+      def port
+        @port ||= pick_port_above(8180)
+      end
+
+      def bind
+        if WatirSpec.platform == :windows
+          '127.0.0.1'
+        else
+          '0.0.0.0'
+        end
+      end
+
+      def listening?
+        $stderr.puts "trying #{bind}:#{port}..."
+
+        TCPSocket.new(bind, port).close
+        true
+      rescue
+        false
+      end
+
+      def should_run?
+        !running?
+      end
+
+      def running?
+        defined?(@running) && @running
+      end
+
+      private
+
+      def run_in_child_process
+        begin
+          require "childprocess"
+        rescue LoadError => ex
+          raise "please run `gem install childprocess` for WatirSpec on Windows + MRI\n\t(caught: #{ex.message})"
+        end
+
+        path = File.expand_path("../../spec_helper.rb", __FILE__)
+
+        process = ChildProcess.build(WatirSpec.ruby, path)
```

Comment:
The`WatirSpec#ruby` method was removed, and this path no longer exists, so this code won't run on Windows as is.